src/language.rs

```diff
@@ -257,14 +256,29 @@ pub trait SynthLanguage: Language + Send + Sync + Display + FromOp + 'static {
         rhs: &Pattern<Self>,
     ) -> ValidationResult;
 
-    fn run_synth() {
-        match Command::parse() {
-            Command::Synth(params) => {
-                synth::synth::<Self>(params);
-            }
-            Command::Derive(params) => {
-                derive::derive::<Self>(params);
-            }
-        }
+    fn run_workload(workload: Workload, prior_rules: Ruleset<Self>) -> Ruleset<Self> {
+        synth::synth(SynthParams {
+            workload,
+            prior_rules,
+            iter_limit: 2,
+            time_limit: 60,
+            node_limit: 300000,
+        })
+    }
+
+    fn run_workload_with_limits(
+        workload: Workload,
+        prior_rules: Ruleset<Self>,
+        iter_limit: usize,
+        time_limit: u64,
+        node_limit: usize,
+    ) -> Ruleset<Self> {
+        synth::synth(SynthParams {
+            workload,
+            prior_rules,
+            iter_limit,
+            time_limit,
+            node_limit,
+        })
```

Comment:
You may want some kind of builder pattern or maybe just a `Config` struct of some kind with defaults to allow an easier time configuring ruler runs. It might look something like:
```rust
T::build_run().with_iter_limit(5000).run_workload(workload)
// or 
T::run_workload(workload, Config {
  rules: vec![ /* my prior rules */ ],
  ..Config::default()
)
```